File: security-micro/src/main/java/pl/damiankaplon/entity/Account.java
package pl.damiankaplon.entity;

import io.quarkus.mongodb.panache.common.MongoEntity;
import lombok.Getter;
import lombok.Setter;
import org.bson.types.ObjectId;

import java.util.Set;

import static pl.damiankaplon.service.SecurityService.Registration;

@MongoEntity(database = "database-security", collection = "accounts")
@Getter
@Setter

public class Account {
    private ObjectId id;
    private String email;
    private String password;
    private String name;
    private String surname;
    private Set<String> groups;
    public static Account from(Registration dto) {
        Account account = new Account();
        account.setEmail(dto.email());
        account.setPassword(dto.password());
        account.setSurname(dto.surname());
        account.setName(dto.name());
        return account;
    }
}

File: security-micro/src/main/kotlin/pl/damiankaplon/entity/Account.kt
package pl.damiankaplon.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import java.util.*
import javax.persistence.Entity
import javax.persistence.Table
import javax.persistence.UniqueConstraint

@Entity
@Table(
    uniqueConstraints = [
        UniqueConstraint(columnNames = ["email"])
    ])
internal class Account: PanacheEntity() {

    lateinit var uuid: UUID
    lateinit var email: String
    lateinit var name: String
    lateinit var surname: String
    lateinit var password: String
    lateinit var accesses: String
    companion object: PanacheCompanion<Account> {
        fun findByEmail(email: String) = find("email", email).firstResult()
        fun findByUUID(uuid: UUID) = find("uuid", uuid).firstResult()
        fun save(account: Account) = persist(account)
    }
}